src/components/report-viewer/report-viewer.tsx

```diff
@@ -34,68 +25,50 @@ export default function ReportViewer({ report, reportType }: ReportViewerProps)
-                labelIconColor={report.severity.colorName}
-                key={report.id}
-                sx={styles.treeItem}
-                nodeId={report.id}
-            >
-                {report.subReports.map((value: ReportTreeType) => initializeTreeDataAndComponent(value))}
-            </ReportItem>
-        );
+    const mapReportsById = useCallback((item: ReportTree) => {
+        reportTreeMap.current[item.id] = item;
+        for (let subReport of item.subReports) {
+            mapReportsById(subReport);
+        }
     }, []);
 
     useEffect(() => {
-        const reportTree = mapReportsTree(report);
-        treeView.current = initializeTreeDataAndComponent(reportTree);
-        setExpandedTreeReports([report.id]);
-        setSelectedReportId(report.id);
-        setSelectedReportType(reportTreeData.current[report.id]?.type);
-    }, [report, initializeTreeDataAndComponent, dispatch]);
+        mapReportsById(reportTree);
+        setExpandedTreeReports([reportTree.id]);
+        setSelectedReportId(reportTree.id);
+        setSelectedReportType(reportTreeMap.current[reportTree.id]?.type);
+    }, [reportTree, dispatch, mapReportsById]);
 
     const handleReportVerticalPositionFromTop = useCallback((node: HTMLDivElement) => {
         setReportVerticalPositionFromTop(node?.getBoundingClientRect()?.top);
     }, []);
 
-    const handleSelectNode = (_: SyntheticEvent, reportId: string) => {
-        if (selectedReportId !== reportId) {
-            setSelectedReportId(reportId);
-            setSelectedReportType(reportTreeData.current[reportId].type);
-        }
-    };
-
-    // The MUI TreeView/TreeItems use useMemo on our items, so it's important to avoid changing the context
-    const isHighlighted = useMemo(
-        () => ({
-            isHighlighted: (reportId: string) => highlightedReportId === reportId,
-        }),
-        [highlightedReportId]
-    );
-
     const onLogRowClick = (data: ReportLog) => {
         setExpandedTreeReports((previouslyExpandedTreeReports) => {
             let treeReportsToExpand = new Set(previouslyExpandedTreeReports);
             let parentId = data.parentId;
-            while (reportTreeData.current[parentId]?.parentId) {
-                parentId = reportTreeData.current[parentId].parentId as string;
+            while (reportTreeMap.current[parentId]?.parentId) {
+                parentId = reportTreeMap.current[parentId].parentId as string;
                 treeReportsToExpand.add(parentId);
             }
             return Array.from(treeReportsToExpand);
         });
         setHighlightedReportId(data.parentId);
     };
 
+    const handleSelectedItem = useCallback(
+        (report: ReportItem) => {
+            if (selectedReportId !== report.id) {
+                setSelectedReportId(report.id);
+                setSelectedReportType(reportTreeMap.current[report.id].type);
+            }
+        },
+        [selectedReportId]
```

Comment:
maybe in a separate PR we can try to refactor this code a bit. I find it weird for example that we go to the extend of using a ref for the map of ids to reports to avoid having it in the dependencies of this callback, but at the same time it has selectedReportId in its dependencies which presumably changes more often than the reportTree (and maybe even always changes when the reporttree is changed?).
Another weird thing is top use useCallback for callbacks for VirtualizedTreeview, but not for callbacks for LogTable (onLogRowClick is not a useCallback).. To be cleaned up later I guess